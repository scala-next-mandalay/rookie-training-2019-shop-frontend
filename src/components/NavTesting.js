import React from 'react';
import PropTypes from 'prop-types';
import {Grid,TextField} from '@material-ui/core';
import '../assets/style.css';
import Hidden from "@material-ui/core/Hidden";
import clsx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
 root: {
    display: 'flex',
    flexWrap: 'wrap',
     
  },
  margin: {
    margin: theme.spacing(1),
     
  },
  textField: {
    flexBasis: 100,
    
  },
 
}));
const NavTesting = ({ categories, setCategoryId, handleDrawerClose }) => {
  const classes = useStyles();


  return (
    <Hidden only={["sm", "xs","md"]} implementation="css">
    <Grid container>
      <Grid item xs={12} align="center">
      <div className="root">
      <TextField
        select
        className={clsx(classes.margin, classes.textField)}
        SelectProps={{
          native: true,
          MenuProps: {
            className: classes.menu,
          },
        }}
        margin="dense"
        variant="outlined"
      > 
      
        {categories.map(obj => (
          <option key={obj.id} onClick={()=>{ setCategoryId(obj.id); }} className="text2">
            {obj.name}
          </option>
        ))}
      </TextField>
      </div>
     
      </Grid>
      </Grid>
    </Hidden>
  );
};

NavTesting.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired).isRequired,
  setCategoryId: PropTypes.func,
  handleDrawerClose: PropTypes.func,
};

export default NavTesting;
