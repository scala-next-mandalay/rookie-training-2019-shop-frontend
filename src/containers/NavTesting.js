import { connect } from 'react-redux'
import NavTesting from '../components/NavTesting'
import { setCategoryId } from '../modules/items'

export default connect(
  (state) => ({
    categories: state.categories.rows,
  }),
  (dispatch) => ({
    setCategoryId: (id) => dispatch(setCategoryId(id)),
  })
)(NavTesting)
