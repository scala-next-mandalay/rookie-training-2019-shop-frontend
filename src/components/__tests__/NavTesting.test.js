import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import { act } from 'react-dom/test-utils';
import NavTesting from '../NavTesting';
/*global expect*/
/*global jest*/
/*global MouseEvent*/

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});

let categories = [{ id: 1, name: "category1" },
                  { id: 2, name: "category2" }];
let id=null;
const setCategoryId = jest.fn((value) => {
  id = value;
});

describe("NavTesting component", () => {
  it('matches the snapshot', () => {
    const NavTestingSnapshot = renderer.create(
      <Parent><NavTesting setCategoryId={setCategoryId} categories={categories}/></Parent>
    ).toJSON();
    expect(NavTestingSnapshot).toMatchSnapshot();
  });
})

describe("testing category", () => {
      it('Test selection', () => {
        let categories = [ {id:1 , name : "category1"},{id :2, name: "category2"}];
        
        act(() => {
          ReactDOM.render((
            <Parent>
              <NavTesting categories={categories} setCategoryId={setCategoryId}/>
            </Parent>
          ), container);
       
          const selectArr = document.querySelectorAll('select');
          act(() =>{
            // selectArr[0][1].value=1
            selectArr[0][1].dispatchEvent(new MouseEvent('click', {bubbles: true}));
          });
         
          //console.log("selectArr[0][1]",selectArr[0][1].textContent)
          expect(setCategoryId).toHaveBeenCalled();
          expect(id).toBe(2);
          
           act(() =>{
            selectArr[0][0].value=1
            selectArr[0][0].dispatchEvent(new MouseEvent('click', {bubbles: true}));
          });
          expect(setCategoryId).toHaveBeenCalled();
          expect(id).toBe(1);
          expect(selectArr[0][0].textContent).toBe("category1");
        });
      });
});
      