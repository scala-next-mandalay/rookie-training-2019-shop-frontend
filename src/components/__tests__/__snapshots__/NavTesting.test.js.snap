// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTesting component matches the snapshot 1`] = `
<div
  className="PrivateHiddenCss-onlySm-7 PrivateHiddenCss-onlyXs-4 PrivateHiddenCss-onlyMd-10"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      align="center"
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-margin-2 makeStyles-textField-3 MuiFormControl-marginDense"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <select
              aria-invalid={false}
              className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSelect MuiOutlinedInput-inputSelect MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense MuiSelect-outlined"
              disabled={false}
              multiple={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
            >
              <option
                className="text2"
                onClick={[Function]}
              >
                category1
              </option>
              <option
                className="text2"
                onClick={[Function]}
              >
                category2
              </option>
            </select>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-176 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-177"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
